stargate/crates/stargate-lib/src/request_pipeline/service_definition.rs

```diff
@@ -35,13 +35,31 @@ impl Service for ServiceDefinition {
             variables: Some(Map::from_iter(variables.into_iter()).into()),
         };
 
+        let headers = &context.request_context.header_map;
+
+        let mut request_builder = surf::post(&self.url).header("userId", "1");
+        for (&name, &value_bytes) in headers.into_iter() {
+            match std::str::from_utf8(value_bytes) {
+                Ok(value) => {
+                    request_builder = request_builder.header(name, value);
```

Comment:
Looks like `.header`'s 2nd argument's type is `impl ToHeaderValues` which means it can accept any struct that implements that trait.
One of the implementors is `HeaderValue` from `http_types`. That struct has a `from_bytes` method, which is the value in the HashMap we have in RequestContext. That seems like the most appropriate way to pass that in.